python/stack_plots.py

```diff
@@ -294,30 +372,142 @@ def draw_plot(plotname="fatjet_msoftdrop", title="myTitle", log=True, compare_da
 
     ROOT.gPad.RedrawAxis()
 
-    latex.DrawLatex(0.9, 0.92+0.03, yearenergy);
-    latexCMS.DrawLatex(0.11,0.92+0.03,"CMS");
-    latexCMSExtra.DrawLatex(0.22,0.92+0.03, cmsExtra);
+    expoffset=0.03
+    if log==True or 1.1*histMax<1000.0:
+        expoffset=0
+    latex.DrawLatex(0.9, 0.92+expoffset, yearenergy);
+    latexCMS.DrawLatex(0.11,0.92+expoffset,"CMS");
+    latexCMSExtra.DrawLatex(0.22,0.92+expoffset, cmsExtra);
 
     if "cutflow" not in plotname:
-        whichsel = plotname.split("_")[len(plotname.split("_"))-1]
+        whichnb  = plotname.split("_")[len(plotname.split("_"))-1]
+        whichmll = plotname.split("_")[len(plotname.split("_"))-2]
+        whichsel = plotname.split("_")[len(plotname.split("_"))-3]
+        ts = 0
         for s in range(0,nsel[whichsel]+1):
-            latexSel.DrawLatex(0.3, 0.87-s*0.025, sels[s])
+            if '1p' not in whichnb and s==7:
+                continue
+            if 'inclusive' not in whichmll and s==8:
+                continue
+            ts = ts+1
+            latexSel.DrawLatex(0.3+3*legoffset, 0.89-ts*(0.028-legoffset), sels[s])
+        if '1p' not in whichnb and nsel[whichsel]>=9:
+            ts = ts+1
+            latexSel.DrawLatex(0.3+3*legoffset, 0.89-ts*(0.028-legoffset), nbbin[whichnb])
+        if 'inclusive' not in whichmll and nsel[whichsel]>=7:
+            ts = ts+1
+            latexSel.DrawLatex(0.3+3*legoffset, 0.89-ts*(0.028-legoffset), mllbin[whichmll])
 
     #print and save
     extension = ""
-    if compare_data:
+    if plotData:
         extension = extension+"_mc+data"
     else:
         extension = extension+"_s+b"
     if log:
-        extension = extension+"_log"
-    else:
-        extension = extension+"_linear"
+        extension = extension+"_logY"
     if args.shape:
         extension = extension+"_areaNormalized"
     
     canvas.SaveAs(args.outDir + plotname + extension + ".png")
 
+    if("pt" in plotname):
```

Comment:
I think we were asked to do the same for mll plots as well?

It would be nice if this became a function (to reduce clutter within the main function) but this is really a second/third order improvement.

Reply:
I have added "mll_pf" to the `if` condition.
For the function, we agreed it may come later.